pallene/checker.lua

```diff
@@ -549,19 +598,43 @@ function Checker:check_stat(stat)
     elseif tag == "ast.Stat.If" then
         stat.condition = self:check_exp_synthesize(stat.condition)
         check_type_is_condition(stat.condition, "if statement condition")
-        self:check_stat(stat.then_)
-        self:check_stat(stat.else_)
+        self:check_stat(stat.then_, false)
+        self:check_stat(stat.else_, false)
 
     elseif tag == "ast.Stat.Break" then
         -- ok
 
+    elseif tag == "ast.Stat.Func" then
+
+        local decl = stat.decl
+        local fname_exp = stat.name
+        decl._type = self:from_ast_type(decl.type)
+        self:check_funcname(fname_exp, decl)
+        stat.value =
+            self:check_exp_verify(stat.value, decl._type, "toplevel function")
```

Comment:
Does this fit in 100 columns without the line break?

Reply:
I don't think it does.